File: CSharpFundamentals/CSharpOOPbasics/WorkingWithAbstractionLab/PointInRectangle/Rectangle.cs
namespace PointInRectangle
{
    public class Rectangle
    {
        private Point topLeft;
        private Point bottomRight;

        public Point BottomRight
        {
            get { return bottomRight; }
            set { bottomRight = value; }
        }

        public Point TopLeft
        {
            get { return topLeft; }
            set { topLeft = value; }
        }        

        public bool Contains(Point point)
        {
            return point.X >= this.TopLeft.X &&
                   point.X <= this.BottomRight.X &&
                   point.Y <= this.BottomRight.Y &&
                   point.Y >= this.TopLeft.Y;                            
        }
    }
}

File: JavaFundamentals/JavaOOP/Working-with-Abstraction-Lab/src/PointInRectangle/Rectangle.java
package PointInRectangle;

public class Rectangle {
    private Point bottomLeft;
    private Point topRight;

    public Point getBottomLeft() {
        return bottomLeft;
    }

    public void setBottomLeft(Point bottomLeft) {
        this.bottomLeft = bottomLeft;
    }

    public Point getTopRight() {
        return topRight;
    }

    public void setTopRight(Point topRight) {
        this.topRight = topRight;
    }

    public boolean contains(Point point) {
        return isXInside(point) && isYInside(point);
    }

    private boolean isXInside(Point point) {
        return point.getX() >= this.bottomLeft.getX() && point.getX() <= this.topRight.getX();
    }

    private boolean isYInside(Point point) {
        return point.getY() >= this.bottomLeft.getY() && point.getY() <= this.topRight.getY();
    }
}
